tutorials/plot_spatial.py

```diff
@@ -11,40 +11,237 @@
+
+# Look at the range of our x and y positions
+ranges = get_pos_ranges(position)
+print(f'The x-position ranges from {ranges[0][0]} to {ranges[0][1]}')
+print(f'The y-position ranges from {ranges[1][0]} to {ranges[1][1]}')
+
+# Plot positions (coordinates marked x are the actual points)
+plt.plot(x_pos, y_pos, '-x', c = 'tab:gray', markersize = 10)
+plt.title('Tracking')
+plt.xlabel('x-position')
+plt.ylabel('y-position')
+plt.legend(['coordinates'])
+
+###################################################################################################
+
+# Compute distance between start and end point
+dist_start_end = compute_distance(x_pos[0], y_pos[0], x_pos[-1], y_pos[-1])
+# Compute distance traveled at each point
+dist_traveled = compute_distances(x_pos, y_pos)
+# Compute total distance traveled
+cumulative_dist_traveled = compute_cumulative_distances(x_pos, y_pos)
+# Compute speed at each point
+bin_widths = np.array([1, 1, 1, 2, 1.5, 1, 1, 1, 2, 1, 1, 1, 1, 1, 0.5])
+speeds = compute_speed(x_pos, y_pos, bin_widths)
+
+###################################################################################################
+
+f, (ax1, ax2, ax3) = plt.subplots(3, 1, sharex=True)
+# Plot distance traveled at each time
+ax1.plot(dist_traveled, '-x', c = 'tab:pink', markersize = 10)
+ax1.set_title('Distance traveled at each point')
+ax1.set_xlabel('time (t)')
+ax1.set_ylabel('distance (u)')
+
+# Plot cumulative distance traveled per time
+ax2.plot(cumulative_dist_traveled, '-x', c = 'tab:olive', markersize = 10)
+ax2.set_title('Cumulative distance traveled at each point')
+ax2.set_xlabel('time (t)')
+ax2.set_ylabel('distance (u)')
+
+# Plot speed at each time point
+ax3.plot(speeds, '-x', c = 'tab:cyan', markersize = 10)
+ax3.set_title('Speed at each point')
+ax3.set_xlabel('time (t)')
+ax3.set_ylabel('speed (u/t)')
+
+# Add some padding between subplots
+f.tight_layout(pad = 0.05)
+# Make figure bigger
+f.set_size_inches((15/2.54, 20/2.54))
+
+###################################################################################################
+# 2. Divide position in spatial bin edges and plot
+# ~~~~~~~~
+#
+# Compute x- and y- spatial bin edges and plot spatial grid. 
+#
+
+###################################################################################################
+
+x_edges, y_edges = compute_spatial_bin_edges(position, bins)
+
+# Compute the width of each spatial bin
+x_bins_spatial_width = get_bin_width(x_edges)
+y_bins_spatial_width = get_bin_width(y_edges)
+print(f'The x spatial bins have width = {x_bins_spatial_width}')
+print(f'The y spatial bins have width = {y_bins_spatial_width}')
 
 ###################################################################################################
 
-# Settings
-...
+# Plot: (i) grid of spatial bins, and (ii) grid of spatial bins with tracking on top
+f, (ax1, ax2) = plt.subplots(1, 2)
+# Plot spatial bins in ax1 and ax2
+for i in range(0, max(len(x_edges), len(y_edges))):
```

Comment:
These plots looks really good! 

In terms of plotting, I think we can again use a plot function here, so that there is not too much plot code directly in the tutorial. Current `plot_positions` allows for plotting bin edges, by using gridlines. This is a bit different from how you are doing it I think, and we could explore a bit on if we prefer one or the other, etc.